lib/trino-orc/src/main/java/io/trino/orc/metadata/statistics/StringStatisticsBuilder.java

```diff
@@ -170,4 +195,73 @@ private Slice dropStringMinMaxIfNecessary(Slice minOrMax)
         }
         return Slices.copyOf(minOrMax);
     }
+
+    static class StringCompactor
+    {
+        public static byte[] computeMin(byte[] bytes, int maxBytes)
+        {
+            if (maxBytes >= bytes.length) {
+                return bytes;
+            }
+            int lastIndex = findLastCharacter(bytes, 0, maxBytes);
+            return Arrays.copyOfRange(bytes, 0, lastIndex);
+        }
+
+        public static byte[] computeMax(byte[] bytes, int maxBytes)
+        {
+            if (maxBytes >= bytes.length) {
+                return bytes;
+            }
+            int lastIndex = findLastCharacter(bytes, 0, maxBytes);
+            int penultimateIndex = findLastCharacter(bytes, 0, lastIndex - 1);
+            int lastCharCodePoint = new String(bytes, penultimateIndex, lastIndex - penultimateIndex).codePointAt(0) + 1;
```

Comment:
Looks expensive.

Reply:
this isn't correct

codepoint+1  may logically overflow. i.e. it won't overflow `int`, as max codepoint ~1M, but it may overflow max codepoint.